tchannel/tornado/connection.py

```diff
@@ -205,50 +210,64 @@ def _loop(self):
         #
         # Must be started only after the handshake has been performed.
         self._loop_running = True
+
         while not self.closed:
```

Comment:
This whole loop was already complicated and hard to refactor and this is just making it worse.